### DISCUSSION TITLE
V4 Sync Roadmap

### DISCUSSION THREAD

**Discussion by zmh-program** [coaidev/coai#285] [Announcements | 公告]
Hi Everyone,

First of all, thank you for your support! 🌟

After in-depth discussions and comprehensive evaluations among the CoAI acquisition team, CoAI product research team, and development team,
we have decided to make **open source** one of our core directions for future development. We will reinvest resources to promote the development of existing open-source projects.

During this process, we are actively exploring solutions that combine enterprise features with open-source commercialization projects to find the best compatibility and solutions for open-source commercialization.
We plan to **select certain features from the existing enterprise and commercial versions and gradually synchronize and adapt them to the open-source version**. Below are the specific features **proposed** to be synchronized and adapted to the open-source version:

## ToC&D Features
1. TTS & STT
2. Plugin System, Script Runtime System
3. Commercial UI (Commercial UI Components, Design, Customizable Themes, etc.)
4. Knowledge Base (Integration with Dify.AI / FastGPT / MaxDB and other RAG solutions, #286 )
5. More Payment Providers, Invitation Mechanism, Profit Management
6. Realtime Voice
7. Self-developed GPTs Platform Open Market
8. Multi API Key Token Distribution & Visitor Credit, Permission Management System
9. Notification System
10. OAuth System

## ToB Features
1. Lightweight API Gateway System
2. Model and Channel Health Monitoring System
3. Cost Analysis, Trend Analysis, User Data Analysis System
4. Inference Task Scheduling and Instance Management System (Ollama/LocalAI Distributed Instance Integration)
5. Model and User Group Rate Limiting (TPM, RPM, RPD) Module
6. User Group System, Token User Distribution Groups
7. Content Moderation System (Supporting Text, Regex Blacklist, Baidu Text Review)

Thank you again for your support! Please stay tuned for our upcoming updates. 🥳 (PS. For specific update progress, u can follow our [discord community](https://discord.gg/rj2tDSmk))

**Comment by kenneth-bro** [comment]
欢呼

**Comment by ikexue** [comment]
so good

**Comment by Cyrene963** [comment]
Chatnio 作者大大万岁！

**Comment by Xheldon** [comment]
有确切的时间点吗？

**Comment by ThorinKong** [comment]
discord进不去了

**Comment by johnzhao2050** [comment]
👍👍👍

**Comment by Pevernow** [comment]
@zmh-program Any update?

**Comment by fenghebin1121** [comment]
作者大大什么时候更新呀，跪求
